odl/operator/default_ops.py

```diff
@@ -171,53 +171,157 @@ def __str__(self):
-            Scalars to multiply ``x[0]`` and ``x[1]`` with, respectively.
+        coeffs : ``space.field`` elements
+            Scalars to multiply ``x[i]`` with, respectively.
 
         Examples
         --------
+        Compute the sum of two vectors:
+
         >>> r3 = odl.rn(3)
-        >>> r3xr3 = odl.ProductSpace(r3, r3)
-        >>> xy = r3xr3.element([[1, 2, 3], [1, 2, 3]])
         >>> z = r3.element()
         >>> op = LinCombOperator(r3, 1.0, 1.0)
-        >>> op(xy, out=z)  # Returns z
-        rn(3).element([2.0, 4.0, 6.0])
+        >>> op([[1, 2, 3],
+        ...     [2, 3, 4]],
+        ...    out=z)  # Returns z
+        rn(3).element([3.0, 5.0, 7.0])
         >>> z
-        rn(3).element([2.0, 4.0, 6.0])
+        rn(3).element([3.0, 5.0, 7.0])
+
+        Can also compute arbitrary weighted sums:
+
+        >>> op = LinCombOperator(r3, 1.0, 2.0, 3.0, 4.0)
+        >>> op([[1, 2, 3],
+        ...     [3, 2, 1],
+        ...     [3, 3, 3],
+        ...     [1, 2, 1]])
+        rn(3).element([20.0, 23.0, 18.0])
         """
-        domain = ProductSpace(space, space)
+        domain = ProductSpace(space, len(coeffs))
         super().__init__(domain, space, linear=True)
-        self.a = a
-        self.b = b
+        self.coeffs = coeffs
 
     def _call(self, x, out=None):
         """Linearly combine ``x`` and write to ``out`` if given."""
         if out is None:
             out = self.range.element()
-        out.lincomb(self.a, x[0], self.b, x[1])
+
+        n = len(self.coeffs)
+
+        if n == 0:
+            out.set_zero()
+        elif n == 1:
+            out.lincomb(self.coeffs[0], x[0])
+        elif n == 2:
+            out.lincomb(self.coeffs[0], x[0], self.coeffs[1], x[1])
+        else:
+            out.lincomb(self.coeffs[0], x[0], self.coeffs[1], x[1])
+            for i in range(2, n):
+                out.lincomb(1, out, self.coeffs[i], x[i])
+
         return out
 
+    @property
+    def adjoint(self):
+        """Return the adjoint.
+
+        Examples
+        --------
+        >>> r3 = odl.rn(3)
+        >>> op = LinCombOperator(r3, 1.0, 1.0)
+        >>> op.adjoint
+        ProductSpaceEmbedding(ProductSpace(rn(3), 2), 1.0, 1.0)
+        """
+        return ProductSpaceEmbedding(self.domain, *self.coeffs)
+
     def __repr__(self):
         """Return ``repr(self)``."""
-        return '{}({!r}, {!r}, {!r})'.format(self.__class__.__name__,
-                                             self.range, self.a, self.b)
+        coeff_str = ', '.join(str(ci) for ci in self.coeffs)
+        return '{}({!r}, {})'.format(self.__class__.__name__,
+                                     self.range, coeff_str)
 
-    def __str__(self):
-        """Return ``str(self)``."""
-        return "{}*x + {}*y".format(self.a, self.b)
+
+class ProductSpaceEmbedding(Operator):
+
+    """Operator mapping a vector to several copies of it::
+
+        ProductSpaceEmbedding(spc, a, b)(x) == [a * x, b * x]
```

Comment:
This is just `BroadcastOperator(*[ScalingOperator(ci) for ci in coeffs])`, no need for a new operator.

Reply:
Well the other fella is `ReductionOperator(*[ScalingOperator(ci) for ci in coeffs])` not convinced we should remove only one of them.

Reply:
But now that you mention it there is a very clear "duality" going on here. Let's think about it for a while

Reply:
You're right. Maybe just make it a function instead?

Reply:
Yes that may actually be the best option. It would also preserve `op.adjoint.adjoint == op` in some sense